src/EnergyPlus/Coils/CoilCoolingDXCurveFitPerformance.hh

```diff
@@ -88,11 +88,11 @@ struct CoilCoolingDXCurveFitPerformance
     void simulate(EnergyPlusData &state,
                   const DataLoopNode::NodeData &inletNode,
                   DataLoopNode::NodeData &outletNode,
-                  int useAlternateMode,
+                  HVAC::CoilMode coilMode,
```

Comment:
This needs some followup. `coilMode` implies this is the current operating mode of the coil, but it is really more like `highestAvailableCoilMode`. it's just being used to know if `CoilCoolingDXCurveFitPerformance::alternateMode` and `alternateMode2` are defined for this coil. 

The actual timestep level operating mode is `CoilCoolingDX::dehumidificationMode`. 

And there are functions like `CoilCoolingDX::getNumModes` that only returns a value of 1 or 2, not 3. Maybe it's not needed where it's currently used, but it's confusing as-is.

Reply:
Okay, what is the concrete suggestion?

1. Change the name of the enumeration from `CoilMode` to `DehumMode`? 
2. Change the name of this variable from `coilMode` to `dehumMode`? 
3. Change something else? 
4. Some combination of 1, 2, and 3?

I am okay with whatever except for implicitly using an enumeration as a boolean.

Reply:
So, `coilMode` is replacing `hasAlternateMode` How about `availableCoilMode` or `maxAvailCoilMode`?

Or just make it `int numModes`.

Reply:
Alrighty.

Reply:
But it is the current mode:

                    if (state.dataCoilCooingDX->coilCoolingDXs[this->m_CoolingCoilIndex].SubcoolReheatFlag) {
                        OperationMode = DataHVACGlobals::coilSubcoolReheatMode;
                    } else if (this->m_DehumidificationMode == 1) {
                        OperationMode = DataHVACGlobals::coilEnhancedMode;
                    }
                    bool const singleMode = (this->m_SingleMode == 1);
                    state.dataCoilCooingDX->coilCoolingDXs[this->m_CoolingCoilIndex].simulate(
                        state, OperationMode, PartLoadFrac, this->m_CoolingSpeedNum, this->m_CoolingSpeedRatio, this->m_FanOpMode, singleMode);

Reply:
OK, I agree with @mjwitte 

    if (!this->performance.original_input_specs.base_operating_mode_name.empty() &&
        !this->performance.original_input_specs.alternate_operating_mode_name.empty() &&
        !this->performance.original_input_specs.alternate_operating_mode2_name.empty()) {
        this->SubcoolReheatFlag = true;
    }